packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -468,7 +509,11 @@ export async function launch({
    */
   function makeRunSwingset(blockHeight, runPolicy) {
     let runNum = 0;
-    async function runSwingset(_phase) {
+    async function runSwingset(phase) {
+      if (phase === CLEANUP) {
+        const allowCleanup = runPolicy.startCleanup();
+        if (!allowCleanup) return false;
```

Comment:
I keep wondering if we have the right layering here. I know @warner would like for `controller.run()` to automatically include cleanup for "simple hosts", but it feels like we're jumping through hoops just to enable non cosmic-swingset embedder.

An imperative `controller.performCleanup()` would be so much cleaner. We could simply call it after we've returned from `processBlockActions`: has the policy done anything? no, cool, then start cleanup with a specific cleanup policy.

Reply:
It's true, I don't want to add API burden to hosts that don't need rate limiting, but I agree that there are some significant hoops here.

Adding `controller.performCleanup(budget)` wouldn't be too hard, it might be a wrapper around `controller.run()` that uses an internally-built `runPolicy`. If we added that, then the cosmic-swingset runPolicy could have a `allowCleanup: () => false`, which is a cheap way to say "I'm going to drive cleanup manually, thanks very much".

OTOH, that API assumes that we're ok with arbitrarily large computron usage during the cleanup runs. In practice, that's probably ok: the GC action deliveries are generally trivial (they just change some refcounts), and the BOYD is usually the last thing that runs so it's not like a computron limit would actually stop any deliveries from happening.

But if we wanted to be consistent with normal runs, we'd need a way to provide a limit, which would mean `performCleanup()` would need to get a host-provided RunPolicy, and merging some automatically-built one with the host-provided one sounds more messy than having the host manage everything in the first place.